cli/internal/context/context.go

```diff
@@ -283,35 +341,38 @@ func GetTargetsFromArguments(arguments []string, configJson *fs.TurboConfigJSON)
 func (c *Context) populateTopologicGraphForPackageJson(pkg *fs.PackageJSON) error {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
+	depMap := make(map[string]string)
 	internalDepsSet := make(dag.Set)
-	depSet := make(dag.Set)
+	externalUnresolvedDepsSet := make(dag.Set)
 	externalDepSet := mapset.NewSet()
 	pkg.UnresolvedExternalDeps = make(map[string]string)
-	for dep := range pkg.Dependencies {
-		depSet.Add(dep)
+
+	for dep, version := range pkg.Dependencies {
+		depMap[dep] = version
 	}
 
-	for dep := range pkg.DevDependencies {
-		depSet.Add(dep)
+	for dep, version := range pkg.DevDependencies {
+		depMap[dep] = version
 	}
 
-	for dep := range pkg.OptionalDependencies {
-		depSet.Add(dep)
+	for dep, version := range pkg.OptionalDependencies {
+		depMap[dep] = version
 	}
 
-	for dep := range pkg.PeerDependencies {
-		depSet.Add(dep)
+	for dep, version := range pkg.PeerDependencies {
+		depMap[dep] = version
 	}
 
 	// split out internal vs. external deps
-	for _, dependencyName := range depSet.List() {
-		if item, ok := c.PackageInfos[dependencyName]; ok {
-			internalDepsSet.Add(item.Name)
+	for dependencyName, version := range depMap {
+		if IsReferenceToLocalPackage(c.PackageInfos, dependencyName, version) {
```

Comment:
I rolled the code that looks up from the `c.PackageInfos` map into the isInternal check since logically that is what it means for the package to not be found in the map. But I suppose this could also be:
```go
if item, ok := c.PackageInfos[dependencyName]; ok && IsReferenceToLocalPackage(item, version) {
```

Reply:
It's not the biggest deal, but I think I like your proposed alternative better. That would allow isReferenceToLocalPackage to be a more focused function.